wgpu/src/lib.rs

```diff
@@ -1286,10 +1286,20 @@ pub struct RenderPass<'a> {
 /// Corresponds to [WebGPU `GPUComputePassEncoder`](
 /// https://gpuweb.github.io/gpuweb/#compute-pass-encoder).
 #[derive(Debug)]
-pub struct ComputePass {
+pub struct ComputePass<'encoder> {
     id: ObjectId,
     data: Box<Data>,
     context: Arc<C>,
+
+    /// Whether end should be called on drop.
+    ///
+    /// Setting this to false is useful for manual ending and
+    /// for safe conversion of the [`ComputePass::encoder_guard`].
+    call_end_on_drop: bool,
```

Comment:
yes, this makes it trivial now to do add explicit end & discard calls which were requested at some point as well :)
Let's do that in a separate PR!

Reply:
removed again, let's take this separate